### ISSUE TITLE
Shared space transfer learning for analyzing multi-site fMRI data

### ISSUE THREAD

**Issue by vitruvi** [easyfmri/easyfmri#1]
Hi,
Can you give the minimal code to reproduce your implementation for SSTL article?
Requirements for the whole project mostly stands for the preprocessing pipeline but you host the data preprocessed.

Code to reproduce on the preprocessed data would be very helpful.

**Comment by myousefnezhad** [comment]
A) Generating the SSTL method using easy fMRI:

As you can find in the paper, SSTL has two learning steps, viz., generating the common shared space for each site (denoted in the paper using {G}^{(d, S_d)}), and then generating the global shared space (denoted by W) using the concatenated version of all common spaces in the training phase (see Eq. 9).

1.1) In easy fMRI, you first need to generate {G}^{(d, S_d)} using Feature Analysis -> Feature Design -> Functional Alignment -> GPU Hyperalignment. The code is also available at:
https://gitlab.com/easyfmri/easyfmri/-/blob/master/Hyperalignment/GPUHA.py

1.2) As you can find in the paper, producing {G}^{(d, S_d)} is an unsupervised procedure and you should generate them independently for every site. You simply concatenate all common spaces from the training phase - for this producer, you may want to use easy Data (Tools -> Data/Result Editor). Please do not forget to scale the concatenated matrix G with zero mean and unit variance. 

2) To generate the global shared space (W), you can use PCA. In easy fMRI, you may go to Feature Analysis -> Feature Design -> Feature Engineering -> PCA. In this part, we use the “sklearn” standard PCA approach --- available at:
https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html

If you are using PCA in easy fMRI, you must put the normalization option on the whole data --- i.e., you can find it in the Parameters -> Input Data Normalization -> Whole Data.

B) As you can see from the SSTL paper, the datasets used were publicly available through OpenNEURO (https://openneuro.org/). The preprocessing parameters and pipeline for each dataset were the same as those proposed in the original papers.
